support-frontend/assets/pages/contributions-landing/components/ContributionFormContainer.jsx

```diff
@@ -96,6 +97,7 @@ function withProps(props: PropTypes) {
     // This is because going 'back' to the /contribute page is not helpful, and the client-side routing would redirect
     // back to /thankyou given the current state of the redux store.
     // The effect is that clicking back in the browser will take the user to the page before they arrived at /contribute
+    storage.setSession('isPaymentComplete', 'true');
```

Comment:
unfortunately this breaks paypal (one off), which requires a redirect. I'm trying to understand how we do preserve some session storage when this happens...

Reply:
Dang. I only tested it using the mock Stripe details locally. Do you want to pair on this in about half an hour? No worries if not!

Reply:
e.g. this session storage does persist through to the thankyou page: https://github.com/guardian/support-frontend/blob/724acf30ad4dd1c0291dd8222e32cd35aba07cb3/support-frontend/assets/pages/contributions-landing/contributionsLandingActions.js#L283
I'm not sure what difference is here

Reply:
Ahhhh....isn't there a function that runs after the paypal redirect?

Reply:
Is it too hacky to do something here? https://github.com/guardian/support-frontend/blob/main/support-frontend/assets/pages/contributions-landing/contributionsLandingInit.js#L179

Reply:
```
dispatch(setGuestAccountCreationToken(window.guardian.guestAccountCreationToken));
dispatch(setThankYouPageStage('thankYouSetPassword'));\
storage.setSession('isPaymentComplete', 'true');
```

Reply:
`guestAccountCreationToken` only applies to a subset of our contributors

Reply:
ahh of course...hmmmm...